cryptol-saw-core/src/CryptolSAWCore/Cryptol.hs

```diff
@@ -2203,6 +2189,15 @@ genCodeForNominalTypes sc nominalMap env0 =
             e <- importExpr sc env fnWithTAbs
             return [(conNm, e)]
 
+-- FIXME: ?
+mapAccumLM :: (Monad m) => (a -> x -> m (a, y)) -> a -> [x] -> m (a, [y])
+mapAccumLM _ acc []     = return (acc, [])
+mapAccumLM f acc (x:xs) =
+  do
+  (acc', y) <- f acc x
+  (acc'', ys) <- mapAccumLM f acc' xs
+  return (acc'', y:ys)
+
```

Comment:
I think this is the same as `mapAccumM :: (Monad m, Traversable t) => (s -> a -> m (s, b)) -> s -> t a -> m (s, t b)` from `Data.Traversable`.